core/client-internal/src/main/java/alluxio/client/file/FileInStream.java

```diff
@@ -258,11 +272,15 @@ private void checkAndAdvanceBlockInStream() throws IOException {
                 mContext.getAluxioBlockStore().getOutStream(currentBlockId, blockSize, address);
           }
         } catch (IOException e) {
-          LOG.warn(BLOCK_ID_NOT_CACHED, currentBlockId, e);
+          if (e.getCause() instanceof BlockAlreadyExistsException) {
```

Comment:
Good call, updated. I kept the setting `mShouldCacheCurrentBlock` outside the function so we do the setting at the same level for all the different branches.